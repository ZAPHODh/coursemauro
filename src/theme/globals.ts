'use client'

import { Montserrat } from 'next/font/google'
import { createGlobalStyle } from 'styled-components'
import { css } from 'styled-components'

const montserrat = Montserrat({
    subsets: ['latin'],
    fallback: ['Georgia', 'ui-serif', 'serif'],
    weight: ['300', '700'],
})

export const GlobalStyles = createGlobalStyle`${() => css`
    html {
        font-family: ${montserrat.style.fontFamily};
    }
    ,
    body {
        padding: 0;
        margin: 0;
    }
    ,
    input {
        font-family: inherit;
    }
    ,
    a {
        color: inherit;
    }
    ,
    * {
        box-sizing: border-box;
    }
    ,
    button {
        font-family: inherit;
    }
`}`
