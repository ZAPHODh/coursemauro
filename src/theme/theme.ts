import { DefaultTheme } from 'styled-components'

export const theme: DefaultTheme = {
    colors: {
        white: '#ffffff ',
        darker: '#0B0B0D',
        dark: '#181619',
        normalDark: '#181B22',
        brightDark: '#272A31',
        gold: '#A76B09',
        brightGold: '#DDA74F',
    },
}
