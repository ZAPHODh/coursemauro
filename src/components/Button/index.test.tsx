import { render } from '@testing-library/react'

import { ThemeClient } from '../../theme/ThemeClient'
import { Button } from './index'

describe('Button', () => {
    it('should renders a Button', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Button>teste</Button>
            </ThemeClient>
        )

        const button = getByRole('button')

        expect(button).toBeInTheDocument()
    })

    it('should renders Button with the correct propperty', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Button width={'150px'} fontSize="inherit">
                    test
                </Button>
            </ThemeClient>
        )

        const button = getByRole('button')
        expect(button).toHaveStyle(`
            width: 150px;
            font-size: inherit;
            `)
    })
    it('should renders Button with the correct children text', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Button>children text</Button>
            </ThemeClient>
        )

        const button = getByRole('button')

        expect(button).toHaveTextContent('children text')
    })

    it('should match snapshot', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Button>test</Button>
            </ThemeClient>
        )

        const button = getByRole('button')

        expect(button).toMatchSnapshot()
    })
})
