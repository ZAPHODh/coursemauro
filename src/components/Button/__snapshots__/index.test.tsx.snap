// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  class="sc-3c650bf1-0 dUazBJ"
  font-size="inherit"
  width="90%"
>
  test
</button>
`;
